### ISSUE TITLE
Built Mac OS X eclipse package won't run?

### ISSUE THREAD

**Issue by dnwe** [ferstl/eclipse-builder#5]
Launching the top-level my-eclipse-name.app:
<img width="532" alt="screen shot 2015-09-18 at 20 14 44" src="https://cloud.githubusercontent.com/assets/8060970/9968789/617f330a-5e42-11e5-9e1e-8712a138805c.png">

Attempting to manually launch ./Contents/Eclipse/Eclipse.app
<img width="532" alt="screen shot 2015-09-18 at 20 14 59" src="https://cloud.githubusercontent.com/assets/8060970/9968792/6a7cd246-5e42-11e5-897b-b7532db80aa7.png">

Interesting the Eclipse.app within the larger package just seems to contain eclipse.ini
i.e.,

Contents/Eclipse/Eclipse.app/
└── Contents
    └── MacOS
        └── eclipse.ini

Have you previously run and used the OS X packages?

**Comment by dnwe** [comment]
(this was eclipse 4.5 btw)

**Comment by ferstl** [comment]
@dnwe Yes, I am actually using my Eclipse distros on a Mac. I just re-tried to create an OS X distro with the `eclipse-mars-example` and it worked fine. I used the latest version of the [Headless Director](http://www.eclipse.org/downloads/download.php?file=/tools/buckminster/products/director_latest.zip) and this command:

```
cd ~/eclipse-builder

./eclipse-builder.sh \
--p2command /home/build/eclipse-build/director/director \
--platform macosx \
--destination /home/build/tmp/macosx-test \
--name macosx-test \
eclipse-mars-example/01_eclipse-ide.conf
```

The directory structure that you described does not look correct. It should look like this:

```
macosx-test-macosx-cocoa-x86_64.app/
  |- Contents/
  |  |- Eclipse/
  |  |  |- features/
  |  |  |- plugins/
  |  |  |- eclipse.ini
  |  |  |- ...
  |  |- MacOS/
  |  |- Resources/
  |  |- Info.plist
```

Have you used recent versions of the P2 director and `eclipse-builder.sh`? The `.app` support was introduced in version 4.5 in June 2015. Previous versions of the P2 director are not able to build correct OS X apps.

**Comment by dnwe** [comment]
Yes it was latest director and the Mars conf set. One difference is that I didn't run the builder on the Mac itsel. The output build was from a Jenkins build running on Linux that generates for all three platforms. I'll see if I can reproduce it

**Comment by ferstl** [comment]
@dnwe I created the distro on a Linux server too. Unfortunately, `eclipse-builder.sh` does not work on OS X due to a very old bash version.

**Comment by dnwe** [comment]
Hmm ok, let me retest and get back to you

**Comment by dnwe** [comment]
I guess the bash4 requirement is due to the use of associative arrays in [eclipse-builder.sh:30](https://github.com/ferstl/eclipse-builder/blob/master/eclipse-builder.sh#L30-L38) ?

**Comment by ferstl** [comment]
Yes it's mostly the associative arrays. I tried to eliminate them more than a year ago and found some other things that behaved differently. But I don't remember anymore what that was (probably something with the `getopt` command).

**Comment by dnwe** [comment]
OK, so I retested this earlier a few times and was able to reproduce the same issue. However, I then fetched director_latest.zip again and re-ran and it correctly generated a build!

It turns out that my version of director _was_ slightly older `'org.eclipse.equinox.p2.director' version='2.3.100.v20140224-1921'` vs `'org.eclipse.equinox.p2.director' version='2.3.200.v20150423-1455'`. So it looks like this is probably user error!
